src/vpn/handler_isVpnAlive.go

```diff
@@ -10,13 +11,18 @@ func (h *Handler) isVpnAlive(serverIp string) bool {
 		return false
 	}
 
-	for i := 0; i < 3; i++ {
-		_, err := exec.Command("ping6", "-c", "1", serverIp).Output()
-		if err != nil {
-			continue
+	for i := 0; i < h.config.VpnCheckRetryCount; i++ {
+		if func() bool {
+			ctx, cancel := context.WithTimeout(context.Background(), h.config.VpnCheckTimeout)
+			defer cancel()
+			_, err := exec.CommandContext(ctx, "ping6", "-c", "1", serverIp).Output()
```

Comment:
Keep in mind that this is going to work only on Unix machines. The Windows alternative is `ping -6`. More general way to write this is to send ICMP packets directly.

Reply:
it is going to be dealt with in windows version